server/ec2spotmanager/tasks.py

```diff
@@ -38,87 +38,90 @@ def check_instance_pool(pool_id):
         if criticalPoolStatusEntries:
             return
 
-        if instance_pool.config.isCyclic() or instance_pool.config.getMissingParameters():
+        if instance_pool.config.provider.isCyclic() or instance_pool.config.provider.getMissingParameters():
             entry = PoolStatusEntry()
             entry.pool = instance_pool
             entry.isCritical = True
-            entry.type = POOL_STATUS_ENTRY_TYPE['config-error']
+            entry.type = EC2_POOL_STATUS_ENTRY_TYPE['config-error']
             entry.msg = "Configuration error."
             entry.save()
             return
 
-        config = instance_pool.config.flatten()
-
-        instances_missing = config.size
-        running_instances = []
-
-        _update_pool_instances(instance_pool, config)
-
-        instances = Instance.objects.filter(pool=instance_pool)
-
-        for instance in instances:
-            instance_status_code_fixed = False
-            if instance.status_code >= 256:
-                logger.warning("[Pool %d] Instance with EC2 ID %s has weird state code %d, attempting to fix...",
-                               instance_pool.id, instance.ec2_instance_id, instance.status_code)
-                instance.status_code -= 256
-                instance_status_code_fixed = True
-
-            if instance.status_code in [INSTANCE_STATE['running'], INSTANCE_STATE['pending'],
-                                        INSTANCE_STATE['requested']]:
-                instances_missing -= 1
-                running_instances.append(instance)
-            elif instance.status_code in [INSTANCE_STATE['shutting-down'], INSTANCE_STATE['terminated']]:
-                # The instance is no longer running, delete it from our database
-                logger.info("[Pool %d] Deleting terminated instance with EC2 ID %s from our database.",
-                            instance_pool.id, instance.ec2_instance_id)
-                instance.delete()
+        config = instance_pool.config.provider.flatten()
+
+        # the vast majority of this is ec2 specific. We will need to add task support for other providers
+        if config.ec2poolconfiguration:
```

Comment:
Remove this diff and all of the diffs in this file that start with "if config.ec2poolconfiguration:" because this is not the right way to add support for other providers. Instead, for now add a hard check that the config is actually EC2 and raise a RuntimeException if it isn't.

Later, you will have to analyze what particular tasks are being done here by this code, abstract them, and move their specialized implementations into the model classes.